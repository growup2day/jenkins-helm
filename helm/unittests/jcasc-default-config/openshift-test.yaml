templates:
  - config-jcasc.yaml
tests:
  - it: should include openshift cluster in default config script when configured
    set:
      domain: dummy.domain
      JCasC:
        defaultConfig: true
        openshiftClouds:
          cs1-dev:
            serverUrl: "https://api.cs1-dev.nz.service.test:6443"
            namespace: "{{ .Release.Namespace }}"
            skipTlsVerify: true
          cs2-dev:
            serverUrl: "https://api.cs2-dev.nz.service.test:6443"
            namespace: "{{ .Release.Namespace }}"
            skipTlsVerify: false
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-openshift.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-openshift.yaml]
  - it: should include blank cloud in default config script when not configured
    set:
      domain: dummy.domain
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-openshift.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-openshift.yaml]
  - it: should not include cloud config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            unclassified:
              openShift:
                clusterConfigs:
                  - name: cs1-dev
                    serverUrl: "https://api.cs1-dev.nz.service.test:6443"
                    defaultProject: "NAMESPACE"
    asserts:
      - isNull:
          path: data.[jcasc-default-config-openshift.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
