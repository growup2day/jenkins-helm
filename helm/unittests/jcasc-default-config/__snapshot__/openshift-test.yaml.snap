should include blank cloud in default config script when not configured:
  1: |
    |-
      unclassified:
        # Configure System > OpenShift Client Plugin
        openShift:
          clusterConfigs: []
should include openshift cluster in default config script when configured:
  1: |
    |-
      unclassified:
        # Configure System > OpenShift Client Plugin
        openShift:
          clusterConfigs:
            - name: cs1-dev
              serverUrl: "https://api.cs1-dev.nz.service.test:6443"
              defaultProject: "NAMESPACE"
              skipTlsVerify: true
            - name: cs2-dev
              serverUrl: "https://api.cs2-dev.nz.service.test:6443"
              defaultProject: "NAMESPACE"
              skipTlsVerify: false
